### PR TITLE
Delete CUDA buildchain from RHODS deployer

### PR DESCRIPTION
**Pull request by atheo89**
Delete CUDA build chain from RHODS deployer

## Description
This PR deletes the CUDA build chain code from odh-deployer repository. The new pre-build images now are going to be loaded into RHODS through ImageStreams objects that will be located `jupyterhub/notebooks/overlays/additional` on odh-manifests repository. The corresponding PR is [here](https://github.com/red-hat-data-services/odh-manifests/pull/263)   

Files that deleted:
- buildchain.sh
- nbc/cuda-11.4.2/manifests.yaml
- From prometheus-configs.yaml (on the 'alerting.rules' section deleted the part of build alerts)

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] For commits that came from upstream, `[UPSTREAM]` has been prepended to the commit message.
- [x] JIRA link(s): https://issues.redhat.com/browse/RHODS-5336
- [ ] The Jira story is acked.
- [x] Live build image:  **quay.io/rh_ee_atheodor/rhods-operator-live-catalog:1.20.0-rhods-5336**
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work.
- [ ] QE contact acknowledges that this has been tested and is approved for merge.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -124,6 +124,9 @@ oc delete crd rhodsquickstarts.console.openshift.io 2>/dev/null || echo "INFO: U
 RHODS_SELF_MANAGED=1
 oc get catalogsource -n openshift-marketplace self-managed-rhods &> /dev/null || RHODS_SELF_MANAGED=0
 
+# If buildconfigs with label rhods/buildchain=cuda-* found, delete them (replaced by pre-uild notebooks).
```
Comment (anishasthana) [inline deploy.sh:127]: is `pre-uild` a typo?

File: deploy.sh [hunk 2]
```diff
@@ -373,7 +376,3 @@ fi
 
 # Add network policies
 oc apply -f network/
-
-# Create the runtime buildchain if the rhods-buildchain configmap is missing,
-# otherwise recreate it if the stored checksum does not match
-$HOME/buildchain.sh
```
Comment (anishasthana) [inline deploy.sh:379]: We should remove this from the dockerfile as well: https://github.com/red-hat-data-services/odh-deployer/blob/main/Dockerfile#L27

Comment (anishasthana) [inline deploy.sh:379]: We will also need to remove https://github.com/red-hat-data-services/odh-deployer/blob/main/Dockerfile#L40 right?

**Review by anishasthana — CHANGES_REQUESTED** [review]
Is there a corresponding rhods-live-builder PR yet for the dockerfile updates?

File: deploy.sh (hunk 1 above)
Comment (atheo89) [reply deploy.sh:127]: yep!

**Comment by atheo89** [comment]
> Is there a corresponding rhods-live-builder PR yet for the dockerfile updates?

Not Yet, I am working on it, I will update all the correspond PR with it when it will be ready

File: Dockerfile [hunk 3]
```diff
@@ -24,7 +24,6 @@ RUN tar -C /usr/local/bin -xvf $TMPDIR/oc.tar.gz && \
     mkdir -p $HOME
 
 COPY deploy.sh $HOME
-COPY buildchain.sh $HOME
 ADD kfdefs $HOME/kfdefs
 ADD monitoring $HOME/monitoring
 ADD consolelink $HOME/consolelink
```
Comment (anishasthana) [inline Dockerfile:29]: We will need to remove the following line too right? 
`ADD nbc $HOME/nbc`

File: deploy.sh [hunk 4]
```diff
@@ -124,6 +124,12 @@ oc delete crd rhodsquickstarts.console.openshift.io 2>/dev/null || echo "INFO: U
 RHODS_SELF_MANAGED=1
 oc get catalogsource -n openshift-marketplace self-managed-rhods &> /dev/null || RHODS_SELF_MANAGED=0
 
+# If buildconfigs with label rhods/buildchain=cuda-* found, delete them (replaced by pre-build notebooks).
```
Comment (samuelvl) [inline deploy.sh:127]:
```suggestion
# TODO: Remove in 1.21
# If buildconfigs with label rhods/buildchain=cuda-* found, delete them (replaced by pre-build notebooks).
```

File: deploy.sh [hunk 5]
```diff
@@ -124,6 +124,12 @@ oc delete crd rhodsquickstarts.console.openshift.io 2>/dev/null || echo "INFO: U
...
+oc delete buildconfig -n redhat-ods-applications -l rhods/buildchain
+
+# If imagestreams with label rhods/buildchain=cuda-* found, delete them (replaced by pre-build notebooks).
```
Comment (samuelvl) [inline deploy.sh:130]:
```suggestion
# TODO: Remove in 1.21
# If imagestreams with label rhods/buildchain=cuda-* found, delete them (replaced by pre-build notebooks).
```

**Comment by lugi0** [comment]
Tested the latest live build (quay.io/rh_ee_atheodor/rhods-operator-live-catalog:1.20.0-rhods-5336) and images are looking good (i.e. cuda is present), still in the process of going through all automated tests.

**Review by samuelvl — APPROVED** [review]
/lgtm

**Pull request merged**